tests/test_flow.py

```diff
@@ -3545,6 +3632,39 @@ def test_none_key_date_is_not_written():
     assert result == expected
 
 
+def test_str_id_with_no_header():
```

Comment:
Please mention csv_source in relevant test names. The ID here is an integer, not a string, so it seems that the test name is wrong (this also seems to apply to multiple tests, not just this one).